app/models/user.rb

```diff
@@ -310,6 +311,14 @@ def regenerate_api_key
     save
   end
 
+  # regenerates sms auth code
+  def regenerate_sms_auth_code
+    begin
+      self.sms_auth_code = Random.alphanum(4)
+    end while User.exists?(sms_auth_code: sms_auth_code)
+    save
```

Comment:
Could you explain why you don't want to validate here? This method has more or less identical behavior to the API key generation which doesn't add `validate: false` there, just curious if I should be changing both, or if you noticed something breaking that shouldn't be due to validation issues.